types/src/data.rs

```diff
@@ -750,27 +764,54 @@ pub fn random_commitment<S: Committable>(rng: &mut dyn rand::RngCore) -> Commitm
         .finalize()
 }
 
-impl<TYPES: NodeType> Committable for ValidatingLeaf<TYPES> {
-    fn commit(&self) -> commit::Commitment<Self> {
-        let mut signatures_bytes = vec![];
-        let signatures = match &self.justify_qc.signatures {
-            YesNoSignature::Yes(signatures) => {
+/// Serialization for the QC assembled signature
+pub fn serialize_signature(signature: &AssembledSignature) -> Vec<u8> {
+    let mut signatures_bytes = vec![];
+    let signatures: Option<
+        (<BLSOverBN254CurveSignatureScheme as SignatureScheme>::Signature,
+            <BitvectorQuorumCertificate<BLSOverBN254CurveSignatureScheme> as
+            QuorumCertificateValidation<BLSOverBN254CurveSignatureScheme>>::Proof)>  = match &signature {
```

Comment:
Wow! Not pretty but it doesn't look like there's a way around it. Nice use of generics and maneuvering the types!